### PR TITLE
[lldb][ClangExpressionParser] Set BuiltinHeadersInSystemModules depending on SDK version

### PR DESCRIPTION
**Pull request by Michael137**
This patch changes the way we initialize `BuiltinHeadersInSystemModules` which is one of the flags controlling Clang's behaviour when the Darwin module is split into more fine-grained modules. The ClangExpressionParser currently unconditionally sets `-fbuiltin-headers-in-system-modules` when evaluating expressions with the `target.import-std-module` setting. This flag should, however, be set depending on the SDK version (which is what the Clang Darwin toolchain does).

Unfortunately, the compiler instance that we create with `ClangExpressionParser` never consults the Clang driver, and thus doesn't correctly infer `BuiltinHeadersInSystemModules`. Note, this isn't an issue with the `CompilerInstance` that the `ClangModulesDeclVendor` creates because it uses the `createInovcation` API, which calls into `Darwin::addClangTargetOptions`.

This patch mimicks how `sdkSupportsBuiltinModules` is used in `Darwin::addClangTargetOptions`.

This ensures that the `import-std-module` API tests run cleanly regardless of SDK version.

The plan is to eventually make the `CompilerInstance` construction in `ClangExpressionParser` go through the driver, so we can avoid duplicating the logic in LLDB. But we aren't there yet.

**Implementation**
* We look for the `SDKSettings.json` in the sysroot directory that we found in DWARF (via `DW_AT_LLVM_sysroot`)
* Then parse this file and extract the SDK version number out of it
* Then mimick `sdkSupportsBuiltinModules` from `Toolchains/Darwin.cpp` and set `BuiltinHeadersInSystemModules` based on it

rdar://116490281

### CODE REVIEW

**Review by adrian-prantl — APPROVED** [review]
Thanks, this LGTM, next step should be to replace the use of `include_directories` with `DW_AT_APPLE_sdk`.

**Comment by ian-twilightcoder** [comment]
This really isn't the right approach. `BuiltinHeadersInSystemModules` isn't the only important default that the driver sets up. Can we change the invocation setup to use clang::createInvocation or clang::CompilerInvocation::CreateFromArgs or clang::BuildCompilation? Or is that planned for another change?

File: lldb/source/Utility/XcodeSDK.cpp [hunk 1]
```diff
@@ -259,6 +259,27 @@ bool XcodeSDK::SupportsSwift() const {
   }
 }
 
+bool XcodeSDK::SDKSupportsBuiltinModules(const llvm::Triple &target_triple,
```
Comment (ian-twilightcoder) [inline lldb/source/Utility/XcodeSDK.cpp:262]: This function got changed in https://github.com/llvm/llvm-project/pull/102239

**Comment by Michael137** [comment]
> This really isn't the right approach. `BuiltinHeadersInSystemModules` isn't the only important default that the driver sets up. Can we change the invocation setup to use clang::createInvocation or clang::CompilerInvocation::CreateFromArgs or clang::BuildCompilation? Or is that planned for another change?

Yes I 100% agree that we should use the driver. This has been a FIXME for many years now. But to unblock the current test failures I think this should be fine to land. And when we switch to using the driver properly we'll remove this.

File: lldb/source/Utility/XcodeSDK.cpp (hunk 1 above)
Comment (Michael137) [reply lldb/source/Utility/XcodeSDK.cpp:262]: I already accounted for the version check changes

Comment (ian-twilightcoder) [reply lldb/source/Utility/XcodeSDK.cpp:262]: But you didn't pick up the environment check

Comment (Michael137) [reply lldb/source/Utility/XcodeSDK.cpp:262]: Don't think we really want to support this for non-native environments tbh

**Pull request merged**